spec/browser/connection.test.js

```diff
@@ -1,283 +1,294 @@
 'use strict';
 
 define(['ably', 'shared_helper'], function(Ably, helper) {
-	var exports = {},
-	_exports = {},
-		closeAndFinish = helper.closeAndFinish,
-		monitorConnection = helper.monitorConnection,
-		simulateDroppedConnection = helper.simulateDroppedConnection,
-		transportPreferenceName = 'ably-transport-preference';
-
-	function supportedBrowser(test) {
-		if(document.body.ononline === undefined) {
-			console.log('Online events not supported; skipping connection.test.js');
-			return false;
-		}
+	helper.describeWithCounter('browser/connection', function (expect, counter) {
+		var exports = {},
+		_exports = {},
+			closeAndFinish = helper.closeAndFinish,
+			monitorConnection = helper.monitorConnection,
+			simulateDroppedConnection = helper.simulateDroppedConnection,
+			transportPreferenceName = 'ably-transport-preference';
+
+		function supportedBrowser() {
+			if(document.body.ononline === undefined) {
+				console.log('Online events not supported; skipping connection.test.js');
+				return false;
+			}
 
-		if(!window.WebSocket || !window.localStorage) {
-			console.log('Websockets or local storage not supported; skipping connection.test.js');
-			return false;
+			if(!window.WebSocket || !window.localStorage) {
+				console.log('Websockets or local storage not supported; skipping connection.test.js');
+				return false;
+			}
+
+			// IE doesn't support creating your own events with new
+			try {
+				var testEvent = new Event('foo');
+			} catch(e) {
+				console.log('On IE; skipping connection.test.js');
+				return false;
+			}
+
+			return true;
 		}
 
-		// IE doesn't support creating your own events with new
-		try {
-			var testEvent = new Event('foo');
-		} catch(e) {
-			console.log('On IE; skipping connection.test.js');
-			return false;
+		function eraseSession(name) {
+			window.sessionStorage && window.sessionStorage.removeItem(name);
 		}
 
-		return true;
-	}
+		it('setup_realtime', function(done) {
+			counter.expect(1);
+			helper.setupApp(function(err) {
+				if(err) {
+					expect(false, helper.displayError(err));
+				} else {
+					expect(true, 'app set up');
+				}
+				counter.assert();
+				done();
```

Comment:
Could we perhaps provide a wrapper around `it` that invokes the test function with a `test` object that has `test.expect` and `test.done` to avoid the need for both asserting the counter and calling done in every test?